decompression.go

```diff
@@ -21,9 +21,14 @@ func defaultTarReader(xzReader *xz.Reader) (func() (*tar.Header, error), func()
 }
 
 func decompressTarXz(tarReader func(*xz.Reader) (func() (*tar.Header, error), func() io.Reader), path, extractPath string) error {
+	tempExtractPath, err := os.MkdirTemp("", "embedded_postgres")
```

Comment:
I see a couple of problems here.
- you're not cleaning up this temp dir (minor problem but it's not nice to leave this stuff behind)
- the temp dir is being created in the current working directory

The latter is a problem because rename doesn't work across filesystems, and this temp dir may be in a different fs than `extractPath`.

What you want is to create a temp *file* in the same directory as the final file. Or create the temp dir under that directory.

Reply:
yeah - that make sense - thanks!

I've updated the PR to do both of those:

1. Have the temp locations in the same location as the extract location.
2. Ensure the temp locations are cleaned up when necessary.